styx-client/src/test/java/com/spotify/styx/client/StyxOkHttpClientTest.java

```diff
@@ -0,0 +1,742 @@
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.startsWith;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import static org.junit.internal.matchers.ThrowableMessageMatcher.hasMessage;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.timeout;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.spotify.styx.api.BackfillPayload;
+import com.spotify.styx.api.BackfillsPayload;
+import com.spotify.styx.api.EventsPayload;
+import com.spotify.styx.api.RunStateDataPayload;
+import com.spotify.styx.client.auth.GoogleIdTokenAuth;
+import com.spotify.styx.model.Backfill;
+import com.spotify.styx.model.BackfillInput;
+import com.spotify.styx.model.EditableBackfillInput;
+import com.spotify.styx.model.Event;
+import com.spotify.styx.model.Resource;
+import com.spotify.styx.model.Schedule;
+import com.spotify.styx.model.TriggerParameters;
+import com.spotify.styx.model.TriggerRequest;
+import com.spotify.styx.model.Workflow;
+import com.spotify.styx.model.WorkflowConfiguration;
+import com.spotify.styx.model.WorkflowId;
+import com.spotify.styx.model.WorkflowInstance;
+import com.spotify.styx.model.WorkflowState;
+import com.spotify.styx.model.data.EventInfo;
+import com.spotify.styx.model.data.WorkflowInstanceExecutionData;
+import com.spotify.styx.serialization.Json;
+import java.io.IOException;
+import java.net.URI;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import okhttp3.HttpUrl;
+import okhttp3.Protocol;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+import okio.Buffer;
+import okio.ByteString;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+@RunWith(JUnitParamsRunner.class)
+public class StyxOkHttpClientTest {
+
+  private static final WorkflowConfiguration WORKFLOW_CONFIGURATION_1 = WorkflowConfiguration.builder()
+      .id("bar-wf_1")
+      .dockerImage("busybox")
+      .dockerArgs(Arrays.asList("echo", "hello world"))
+      .schedule(Schedule.DAYS)
+      .build();
+
+  private static final WorkflowConfiguration WORKFLOW_CONFIGURATION_2 = WorkflowConfiguration.builder()
+      .id("bar-wf_2")
+      .dockerImage("busybox")
+      .dockerArgs(Arrays.asList("echo", "hello world"))
+      .schedule(Schedule.DAYS)
+      .build();
+
+  private static final Workflow WORKFLOW_1 = Workflow.create("f[ ]o-cmp", WORKFLOW_CONFIGURATION_1);
```

Comment:
Do we allow non these characters in component and workflow id? Do we want to?

Reply:
I've seen them in the wild at least so I guess so.